assetconnection/opcua/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/assetconnection/opcua/conversion/ValueConverter.java

```diff
@@ -135,16 +137,28 @@ public Variant convert(TypedValue<?> value, NodeId targetType) throws ValueConve
             }
             BuiltinDataType builtinDataType = BuiltinDataType.fromNodeId(targetType);
             if (value.getValue() != null && Objects.equals(builtinDataType.getBackingClass(), value.getValue().getClass())) {
+                if ((value.getDataType() == Datatype.DATE_TIME) && (targetType == Identifiers.DateTime)) {
+                    return new Variant(new DateTime(((ZonedDateTime) value.getValue()).toInstant()));
+                }
                 return new Variant(value.getValue());
             }
+            if ((value.getDataType() == Datatype.DATE_TIME) && (targetType.equals(Identifiers.DateTime))) {
+                return new Variant(new DateTime(((ZonedDateTime) value.getValue()).toInstant()));
+            }
             return new Variant(ImplicitConversions.convert(value.getValue(), builtinDataType));
         }
 
 
         @Override
         public TypedValue<?> convert(Variant value, Datatype targetType) throws ValueConversionException {
             try {
-                return TypedValueFactory.create(targetType, value.getValue().toString());
+                if ((targetType == Datatype.DATE_TIME) && (value.getValue() instanceof DateTime)) {
```

Comment:
Does it make sense to move this functionality to the TypedValueFactory.create method so that other components can use these functionality as well?

Reply:
I don't think that makes sense. 
The DateTime value here is a specific type of Milo. I don't think that it's useful for other components. 
Addititionally, a dependency from the model to Milo would be necessary, what also doesn't make sense in my opinion.